tortoise/expressions.py

```diff
@@ -215,10 +215,13 @@ def __init__(self, query: "AwaitableQuery") -> None:
         self.query = query
 
     def get_sql(self, **kwargs: Any) -> str:
-        return self.query.as_query().get_sql(**kwargs)
+        self.query._choose_db_if_not_chosen()
+        return self.query._make_query(**kwargs)[0]
```

Comment:
It's seems unclear - what exactly are pypika_kwargs and what can be passed there

Because as I am reading code - it uses ambigious `**pypika_kwargs` everywhere, even though only parameter passed I see explicitly used - is parametrizer 

What else can be there, that is passed further down to `get_sql`? 

Are we able to give more explicit typing to such params?

Reply:
@abondar the things that could be in kwargs: dialect, with_namespace, secondary_quote_char, alias_quote_char, as_keyword, etc. Basically it holds the context when pypika builds SQL.

I'm not fond of this either but this is a legacy of pypika, you can search for `get_sql` there to see what I mean. Since `Subquery` extends `Term` it needs to use this approach. Same for `_make_query` in `queryset.py` but I guess the issue became more obvious since I renamed `kwargs` to `pypika_kwargs` there. This PR does not change anything fundamentally about this approach. I think we can improve this by doing something similar as `ResolveContext` but I'm not sure that this PR is a good place for it since it's already quite large.